staging/src/kubevirt.io/api/roundtrip/README.md

```diff
@@ -0,0 +1,21 @@
+### Description:
+
+1. This creates JSON and YAML files for all the API exposed by kubevirt in group-version "kubevirt.io/v1", 
```

Comment:
I'm likely missing something here but why do we need this when we have https://github.com/kubevirt/api ?

Reply:
@lyarwood we need it in-tree because the idea is that these tests will be plugged into CI so APK reviewers can investigate the failure of the test for backwards compatibility breakage. If we do it in kubevirt/api repo, the test will be run postmerge so it will defeat the - purpose of helping API reviewers. Besides contents of the kubevirt/api repo are also in-tree in the staging dir

Reply:
@alaypatel07 right apologies for the confusion about *also* running tests in that repo, that would be pointless if we ran them against the original PR in kubevirt/kubevirt already. However IMHO my point still remains that we don't need to generate yet more yaml/json in this repo to test the backward compatibility of a PR when we already have versioned copies of the API in https://github.com/kubevirt/api .

Reply:
@lyarwood sorry I was missing your point. The versioned copy of API unfortunately does not provide us with a guarantee that API object will be backward compatible. For example:
the https://github.com/kubevirt/api/blob/main/core/v1/types.go#L86 this field could be changed to an int type in this release as follows and there are currently no automated tests to prevent that while it is clearly a backward incompatible change
```
	PriorityClassName int `json:"priorityClassName,omitempty"`
```

Here is a real example that could be caught by this automation: https://github.com/kubevirt/kubevirt/pull/6709/files#diff-34a591e3aa47260cfad791064c20f34f98fdca4a63532fbf7184e68682f4fcb8R254, apologies for picking on this PR. Notice the field has `+optional` tag but missing `omitempty` in its json tags

Reply:
I'm not suggesting that we don't need to write tests to pickup on these incompatible changes just the need to generate duplicate data in this repo when it already exists elsewhere. Could we not use the copies of the API in https://github.com/kubevirt/api to detect such a change?

Reply:
I've spoken to @alaypatel07 on Slack about this and I unfortunately missed that this is common practice in core k8s. The alternative of hosting these generated manifests elsewhere also isn't simple so for now I'm going to remove my concern here.  Thanks again for taking the time to talk about this @alaypatel07.